server/handler/base.go

```diff
@@ -100,19 +134,15 @@ func (b *Base) EvaluateFetchedConfig(ctx context.Context, mbrCtx pull.Membership
 
 	if fetchedConfig.Invalid() {
 		logger.Warn().Err(fetchedConfig.Error).Msgf("invalid policy: %s", fetchedConfig)
-
-		s := b.MakeStatus("error", fetchedConfig.Description(), pr)
-		_, _, err := client.Repositories.CreateStatus(ctx, fetchedConfig.Owner, fetchedConfig.Repo, srcSHA, s)
+		err := b.PostStatus(ctx, client, fetchedConfig.Owner, fetchedConfig.Repo, srcSHA, "error", fetchedConfig.Description(), pr)
```

Comment:
Looking at this in context, another design might be:

```go
status := b.MakeStatus(...)
b.PostStatus(ctx, client, owner, repo, sha, s)
```

`PostStatus` would clone the status object and change the context as needed.

Reply:
feel like that gives you the impression you can post your own status that you made with `MakeStatus`. since we always call both, feels simpler to have it be one method.